packages/vscode-extension/src/webview/providers/ApplicationRootsProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import { PropsWithChildren, useContext, createContext, useState, useEffect, useMemo } from "react";
+import { makeProxy } from "../utilities/rpc";
+import {
+  ApplicationRoot,
+  AppRootConfig as AppRootConfigInterface,
+} from "../../common/AppRootConfig";
+import { EasBuildConfig } from "../../common/EasConfig";
+
+const appRootConfigProxy = makeProxy<AppRootConfigInterface>("LaunchConfig");
+
+export interface AppRootConfig {
+  xcodeSchemes: string[];
+  easBuildProfiles: EasBuildConfig;
+}
+
+export function useAppRootConfig(appRootFolder: string | undefined): AppRootConfig {
+  const [xcodeSchemes, setXcodeSchemes] = useState<string[]>([]);
+  const [easBuildProfiles, setEasBuildProfiles] = useState<EasBuildConfig>({});
+
+  useEffect(() => {
+    if (appRootFolder === undefined) {
+      setXcodeSchemes([]);
+      setEasBuildProfiles({});
+      return;
+    }
+    appRootConfigProxy.getAvailableEasProfiles(appRootFolder).then(setEasBuildProfiles);
+    appRootConfigProxy.getAvailableXcodeSchemes(appRootFolder).then(setXcodeSchemes);
+  }, [appRootFolder]);
+
+  return useMemo(
+    () => ({
+      xcodeSchemes,
+      easBuildProfiles,
+    }),
+    [xcodeSchemes, easBuildProfiles]
+  );
+}
+
+const ApplicationRootsContext = createContext<ApplicationRoot[]>([]);
+
+export default function ApplicationRootsProvider({ children }: PropsWithChildren) {
+  const [applicationRoots, setApplicationRoots] = useState<ApplicationRoot[]>([]);
+
+  useEffect(() => {
+    appRootConfigProxy.getAvailableApplicationRoots().then(setApplicationRoots);
```

Comment:
this won't allow us to get the updated list of app roots ever. Are you sure you want this to work in that way?

Reply:
At this time, it's never updated anyway.
If in the future we decide to update it (by watching the workspace files and searching for the new approots whenever something changes), we will have to change it here as well.